### DISCUSSION TITLE
Trig event when widget player is completely loaded

### DISCUSSION THREAD

**Discussion by krono-i2** [AzuraCast/AzuraCast#5929] [Ideas]
Hello!
It would be useful to trig an event the widget player is completely loaded, in order to dinamically modify the actual DOM structure.
At the moment, when you try to modify the DOM using the custom javascript box, usually the widget elements are not fully loaded.
Is there a way to trig an event when the widget elements are fully loaded?
Thanks for help.

Ivan

**Comment by BusterNeece** [comment]
@krono-i2 Done, in the latest Rolling Release (deploying right now) I added a new event dispatch when the player component is mounted:

```javascript
document.dispatchEvent(new CustomEvent("player-ready"));
```

Should be able to listen to it in custom code via:

```javascript
$(document).on('player-ready', function() {
  // Do a thing
});
```

**Comment by krono-i2** [comment]
Great!
Thanks so much!
